app/models/grda_warehouse/hud/client.rb

```diff
@@ -1785,6 +1785,24 @@ def last_projects_served_by(include_confidential_names: false)
       end.uniq.sort
     end
 
+    def last_intentional_contacts_ignore_visibile_status(include_confidential_names: false, skip_confidential_projects: false, include_dates: false)
+      contacts = processed_service_history&.last_intentional_contacts
+      return [] unless contacts.present?
+
+      projects = GrdaWarehouse::Hud::Project.all.index_by(&:id)
```

Comment:
This will need to go outside of the method call or we'll just load them all for each client.  Also, we should confirm `project_id` from `last_intentional_contacts` is `project.id` not `project.ProjectID`

Reply:
The id should be correct. This method is set up mimicking the last_intentional_contacts method except with the user parameter pulled out.

I'll plan to move the project data up higher in the method stack to clean it up a bit too.

```
def last_intentional_contacts(user, include_confidential_names: false, skip_confidential_projects: false, include_dates: false)
  contacts = processed_service_history&.last_intentional_contacts
  return [] unless contacts.present?

  visible_projects = user.visible_projects_by_id
  visible_project_ids = user.visible_project_ids_enrollment_context
  contacts = JSON.parse(contacts)

  contacts.sort_by { |c| c['date']&.to_date || 5.years.ago }.reverse.map do |contact|
    project_id = contact['project_id']
    next if project_id.present? && !project_id.in?(visible_project_ids)

    project = visible_projects[project_id]
    next if project&.confidential? && skip_confidential_projects

    name = project&.name(ignore_confidential_status: include_confidential_names) || contact['project_name']
    name += ': ' + contact['date']&.to_date.to_s if include_dates
    name
  end.compact.uniq
end
```

```
def visible_projects_by_id
    @visible_projects_by_id ||= GrdaWarehouse::Hud::Project.find(visible_project_ids_enrollment_context).index_by(&:id)
end
```

Reply:
@eanders the project names are being pulled into the method as a parameter based on out call. There were some additional changes needed too as all of the contacts only have the project_id and date. The project names being passed in are having their safe name checked and set ahead of time.